core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -103,34 +122,45 @@ public void close() throws IOException {
      */
     public void abort() throws IOException {
         close();
-        tmpFile.delete();
+        Files.deleteIfExists(tmpFile);
     }
 
     public void commit() throws IOException {
         close();
-        if (destFile.exists()) {
-            try {
-                Util.deleteFile(destFile);
-            } catch (IOException x) {
-                tmpFile.delete();
-                throw x;
-            }
+        try {
+            // Try to make an atomic move.
+            Files.move(tmpFile, destFile, StandardCopyOption.ATOMIC_MOVE);
+        } catch (IOException e) {
```

Comment:
[AtomicMoveNotSupportedException](https://docs.oracle.com/javase/7/docs/api/java/nio/file/AtomicMoveNotSupportedException.html) is a type of [FileSystemException](https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystemException.html) which is a type of IOException.